src/api/app/models/user.rb

```diff
@@ -696,16 +700,7 @@ def delete!
   end
 
   def involved_projects_ids
-    # just for maintainer for now.
-    role = Role.hashed['maintainer']
-
-    ### all projects where user is maintainer
-    projects = relationships.projects.where(role_id: role.id).pluck(:project_id)
-
-    # all projects where user is maintainer via a group
-    projects += Relationship.projects.where(role_id: role.id).joins(:groups_users).where(groups_users: { user_id: id }).pluck(:project_id)
-
-    projects.uniq
+    (direct_maintained_projects.pluck(:project_id) + group_maintained_projects.pluck(:project_id)).uniq
```

Comment:
You mean one pipe ``|``, not two ``||`` ;)

I did a short benchmark which showed that it is actually slower:
```
Benchmark.measure { (1..100).each { u.involved_projects_ids } } 
uniq: 2.180000   0.440000   2.620000 (  8.730327)
pipe: 2.610000   0.670000   3.280000 (  9.440138)
```